osquery/tables/system/linux/md_tables.cpp

```diff
@@ -396,7 +398,8 @@ static inline bool handleMDStatuses(
     const std::string& searchTerm,
     std::function<void(const std::string& line)> handleFunc) {
   std::size_t pos = lines[n + 1].find(searchTerm);
-  if (pos != std::string::npos) {
+  if (pos != std::string::npos &&
+      lines[n + 1].length() > (pos + searchTerm.length())) {
```

Comment:
Like the lines above, this is undefined behavior if `n + 1` is out of bounds for the `std::vector`. These do not act like `std::maps` in that an object could be created then referenced (playing nicely).